stream_batch.go

```diff
@@ -203,14 +209,14 @@ func (sbr *streamBatchReader) getPollWindow() pollWindow {
 	if queryWindowRightEnd.Before(confidenceWindowStart) {
 		return pollWindow{
 			begin: windowStart,
-			end:   gocql.MinTimeUUID(queryWindowRightEnd),
+			end:   gocql.MinTimeUUID(queryWindowRightEnd.Add(sbr.config.Advanced.PostQueryDelay)),
```

Comment:
.. after reading my code multiple times now - I think it was because the window is calculated first, then the delay is resolved and executed. I think this was very weird to understand and maybe some wider refactoring should be done?

see ref: https://github.com/scylladb/scylla-cdc-go/blob/e0b7b88bca47792a8dd560c048c113782776d48e/stream_batch.go#L132-L141

Reply:
I think this was one of those details I only was able to discover while debugging with the added detailed logs. I remember this was confusing and time consuming. ^^

With this in place I was able to achieve and actually see the exact 15s expected latency, as I mentioned in my [4th comment](https://github.com/scylladb/scylla-cdc-go/pull/10#issuecomment-993078436).

Reply:
> .. after reading my code multiple times now - I think it was because the window is calculated first, then the delay is resolved and executed. I think this was very weird to understand and maybe some wider refactoring should be done?

Right, now I see that the delay is wrong. The window is calculated first, then there is a sleep, then the loop goes to the next iteration and uses the query window calculated before the sleep. This is clearly wrong, the window should be used in a query immediately after being computed. The proper fix would be to move `wnd = sbr.getPollWindow()` at the end of the loop.

Reply:
Or - even better - `wnd := sbr.getPollWindow()` should be put at the beginning of the `for` loop and the other calls to `getPollWindow()` removed. I really don't know why it wasn't written like that in the beginning...

Reply:
I also agree that this code could use some refactoring :) Perhaps the logic responsible for tracking per-stream progress and calculating windows could be moved to an abstraction separate from the `streamBatchReader`.

Reply:
yes, current state of the PR could be called a _workaround_ at best 😅, glad I remembered why I've had added this at least. Thanks for pointing out 🧐